app/src/main/java/org/schabi/newpipe/player/playqueue/PlayQueue.java

```diff
@@ -527,6 +527,9 @@ public boolean equals(@Nullable final Object obj) {
         if (size() != other.size()) {
             return false;
         }
+        if (other.getIndex() != getIndex()) {
```

Comment:
This method is also used in other places where checking for the index is not wanted, e.g. in `VideoDetailFragment`. I would completely remove the `equals()` and `hashCode()` methods and create two new methods: `equalStreams(PlayQueue)` and `equalStreamsAndIndex(PlayQueue)`. Then the two methods can be called when needed, providing an easier to read code, and also preventing casting from `Object`.